swarm/src/behaviour.rs

```diff
@@ -561,6 +572,13 @@ pub struct ExternalAddrExpired<'a> {
     pub addr: &'a Multiaddr,
 }
 
+/// [`FromSwarm`] variant that informs the behaviour that a new external address for a remote peer was detected.
+#[derive(Clone, Copy, Debug)]
+pub struct NewExternalAddrOfPeer<'a> {
+    pub peer_id: &'a PeerId,
```

Comment:
PeerId is `Copy` so we don't need a reference to it.

Reply:
:heavy_check_mark: